cocos/ui/button.ts

```diff
@@ -999,6 +999,38 @@ export class Button extends Component {
         }
     }
 
+    private _getSpriteFrameByState (state: State): SpriteFrame | null {
+        switch (state) {
+        case State.NORMAL:
+            return this._normalSprite;
+        case State.DISABLED:
+            return this._disabledSprite;
+        case State.HOVER:
+            return this.hoverSprite;
+        case State.PRESSED:
+            return this._pressedSprite;
+        default:
+            // Should not arrive here.
+            return null;
+        }
+    }
+
+    private _getColorByState (state: State): Color {
+        switch (state) {
+        case State.NORMAL:
+            return this._normalColor;
+        case State.DISABLED:
+            return this._disabledColor;
+        case State.HOVER:
+            return this._hoverColor;
+        case State.PRESSED:
+            return this._pressedColor;
+        default:
+            // Should not arrive here.
+            return new Color();
```

Comment:
Is a warning message necessary?

Reply:
I plan to add warning message, but it will add package size. And i plan to add `assert()`, but it will delete `return new Color();` when saving file in VSCode.

Reply:
May be i can add warning message in` if (DEBUG)`

Reply:
Using the interface inside debug.ts?

Reply:
Fixed